modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -0,0 +1,1200 @@
+        lhs_eq.erase(0, idx + comma.length());
+        lhs_eq_tokens.emplace_back(token);
+    }
+}
+
+
+void LayerEinsumImpl::calculateOutputShape()
+{
+
+    if (outputSize!= 1)
+    {
+        CV_Error(Error::StsError,
+        cv::format("Einsum layer should only have one output, currenly [%d]", outputSize));
+    }
+
+    // Traverse through each of the subscript labels within the output subscript.
+    bool middleOfEllipsis = false;
+    // int64_t ellipsisCharCount = 0;
+
+    subscriptIndicesToOutputIndices.resize(numLetterIndices, -1);
+
+    std::array<int, numOfLetters> outputLetterToCount;
+    outputLetterToCount.fill(0);
+
+    int outputDimCounter = 0;
+    for (auto letter : rhs_eq)
+    {
+        if(letter == '.')
+        {
+            CV_Error(Error::StsNotImplemented, "Ellipsis are not supported yet");
+        } else {
+            if (middleOfEllipsis)
+            {
+                CV_Error(Error::StsError, "Encountered '.' character that is"
+                " not part of output subscript");
+            }
+
+            auto letterIndex = letterToIndex(letter);
+
+            if (letterIndex == -1)
+            {
+                 CV_Error(Error::StsError,
+                    "The only permissible subscript labels are"
+                    " lowercase letters (a-z) and uppercase letters (A-Z).");
+            }
+
+            if (outputLetterToCount[letterIndex] != 0)
+            {
+                CV_Error(Error::StsError,
+                 "Output subscript constains repeated letters");
+            }
+
+            ++outputLetterToCount[letterIndex];
+
+            auto mappedIndex = letter2index[letterIndex];
+
+            if(mappedIndex == -1)
+            {
+                CV_Error(Error::StsError,
+                "Output subscript has letters that were not encountered in the inputs");
+            }
+
+            // Push output dimention
+            // Einsum layer only has one output vector
+            einsumOutDims.emplace_back(subscriptIndicesToDimValue[mappedIndex]);
+
+            // Reset the last input index for this subscript label
+            // given that it is seen in the output and hence can't be reduced
+            subscriptIndicesToLastInput[mappedIndex] = -1;
+            subscriptIndicesToOutputIndices[mappedIndex] = outputDimCounter++;
+        }
+    }
+}
+
+void LayerEinsumImpl::createOutputSubsctipt()
+{
+    // The explicit form requires no operation, as the output
+    // would have already been parsed during the input parsing process.
+    if(explicitEquation)
+    {
+        // Ensure that the provided explicit equation includes an ellipsis if the input contains ellipses.
+        if(numOfEllipsisDims > 0)
+        {
+            if(rhs_eq.find("...") == std::string::npos)
+            {
+                CV_Error(Error::StsError,
+                "Provided output subscript does not include ellipsis while Inputs subscrits constain ellipsis");
+            }
+        }
+    }
```

Comment:
I propose to raise not implemented exception on else branch to catch errors easier.